src/sidebar/services/test/api-test.js

```diff
@@ -156,6 +156,23 @@ describe('APIService', () => {
     return api.search({ uri: 'http://foobar.com/?foo=bar;baz=qux' });
   });
 
+  // Test that covers the most critical use case of multiple values being
+  // provided for an API parameter. Other API calls accept multiple values
+  // as well though.
+  it('repeats query parameters when multiple values are provided', () => {
+    const pdfURL = 'https://example.com/test.pdf';
+    const fingerprintURL = 'urn:x-pdf:foobar';
+
+    expectCall(
```

Comment:
Can you help me understand `expectCall`'s behavior better (the docstring for it wasn't filling me in)? Is it that `api.search` as called subsequently would...throw or something if it didn't match the "expected call" that had been set up? I'm having trouble following the wiring here, and figured I'd use the lazy way out by asking. Put another way, what does a failure state look like?

Reply:
You're right. The `fetch` call rejects if a request doesn't have the expected parameters, which in turn will cause the API method call to reject.